src/modules/design/diagram-detail/diagram-detail.ts

```diff
@@ -215,6 +216,43 @@ export class DiagramDetail {
     });
     const processModelId: string = processModel.id;
 
+    try {
+      await solutionToDeployTo.service.loadDiagram(processModelId);
+
+      this.showDiagramExistingModal = true;
+
+      const modalResult: Promise<boolean> = new Promise((resolve: Function, reject: Function): boolean | void => {
+        const cancelModal: EventListenerOrEventListenerObject = (): void => {
+          this.showDiagramExistingModal = false;
+          resolve(false);
+
+          document.getElementById('cancelDiagramDeploy').removeEventListener('click', cancelModal);
+          // tslint:disable-next-line: no-use-before-declare
+          document.getElementById('overrideDiagramOnSolution').removeEventListener('click', proceedUpload);
+        };
+
+        const proceedUpload: EventListenerOrEventListenerObject = (): void => {
+          this.showDiagramExistingModal = false;
+          resolve(true);
+
+          document.getElementById('cancelDiagramDeploy').removeEventListener('click', cancelModal);
+          document.getElementById('overrideDiagramOnSolution').removeEventListener('click', proceedUpload);
+        };
+
+        setTimeout(() => {
+          document.getElementById('cancelDiagramDeploy').addEventListener('click', cancelModal, {once: true});
+          document.getElementById('overrideDiagramOnSolution').addEventListener('click', proceedUpload, {once: true});
+        }, 0);
+      });
+
+      if (!await modalResult) {
```

Comment:
Dieser Ausdruck negiert das Promise, nicht den erhaltenden Rückgabewert des Promises. 

Um das Ergebnis zu negieren, müsstest du folgendes machen:

```ts
if(!(await modalResult)) {
  [...]
}
```

Mach das besser so:

```suggestion
      const modalResult: boolean = await modalResultPromise;
      if (!modalResult) {
```